menpo/image/base.py

```diff
@@ -2525,17 +2530,38 @@ def pixels_range(self):
 
     def rolled_channels(self):
         r"""
+        Deprecated - please use the equivalent ``as_rolled_channels`` method.
+        """
+        warn('This method is no longer supported and will be removed in a '
+             'future version of Menpo. '
+             'Use .as_rolled_channels() instead.',
+             MenpoDeprecationWarning)
+        return self.as_rolled_channels()
+
+    def as_rolled_channels(self, out_dtype=None):
```

Comment:
What about simply rolled_channels_ndarray?

On 6 June 2016 at 13:24, James Booth <notifications@github.com> wrote:

> In menpo/image/base.py
> <https://github.com/menpo/menpo/pull/702#discussion_r65880732>:
>
> > @@ -2525,17 +2530,38 @@ def pixels_range(self):
> >
> >      def rolled_channels(self):
> >          r"""
> > +        Deprecated - please use the equivalent ``as_rolled_channels`` method.
> > +        """
> > +        warn('This method is no longer supported and will be removed in a '
> > +             'future version of Menpo. '
> > +             'Use .as_rolled_channels() instead.',
> > +             MenpoDeprecationWarning)
> > +        return self.as_rolled_channels()
> > +
> > +    def as_rolled_channels(self, out_dtype=None):
>
> Totally agree with your opening paragraph.
>
> .pixels_array_with_channels_at_back() is definitely too long. I don't
> think we need to specify array as we already have the convention that
> pixels gives access to the underlying numpy array.
>
> Yup, had a similar thought with the .pixels convention already existing
>
> So I would vote for .pixels_with_channels_at_back().
>
> Agreed. Also has the nice benefit that it will appear when tab completing
> .pixels, aiding discoverability.
>
> However, obviously this needs to be a method as it involves a copy.
>
> Agreed.
>
> We also need to fix the init_from_rolled_channels to have consistent
> naming e.g. init_from_channels_at_back().
>
> Yup, agreed. This feels like an overall better solution to a really tricky
> naming issue.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/menpo/menpo/pull/702/files/56bf939db7eb788371c54563876feaaf90bc5a3f#r65880732>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAy0R5GQgndD5V2PoPxjfutQ2RHyHGhdks5qJBFpgaJpZM4Ine8W>
> .
>